assets/js/src/settings/store/normalize_settings.ts

```diff
@@ -0,0 +1,219 @@
+import _ from 'lodash';
+import { t } from 'common/functions';
+import { Settings } from './types';
+
+export default function normalizeSettings(data: any): Settings {
+  const text = asString('');
+  const disabledCheckbox = asBoolean('1', '0', '0');
+  const enabledCheckbox = asBoolean('1', '0', '1');
+  const disabledRadio = asBoolean('1', '', '');
+  const enabledRadio = asBoolean('1', '', '1');
+  const emptyArray = asStringArray([]);
+  const smtpServer = asEnum(['server', 'manual', 'AmazonSES', 'SendGrid'], 'server');
```

Comment:
For enums, shouldn't we use native TS enums? I mean, for instance, using a named type `SettingsSmtpServerEnum` here. Or is that bad since those are internally represented as numbers? 🤔

Reply:
We do use TS enums in `settings/store/types.ts`, but here we need a function that takes any value and forces it to be in the enum.

Reply:
Those are union types in fact, but that's fine, it seems the TS "Enum" would indeed be a number, internally.